frontend/deb/template_control.go

```diff
@@ -0,0 +1,158 @@
+package deb
+
+import (
+	_ "embed"
+	"fmt"
+	"io"
+	"strings"
+	"text/template"
+
+	"github.com/Azure/dalec"
+	"golang.org/x/exp/maps"
+	"golang.org/x/exp/slices"
+)
+
+func WriteControl(spec *dalec.Spec, target string, w io.Writer) error {
+	return controlTmpl.Execute(w, &controlWrapper{spec, target})
+}
+
+type controlWrapper struct {
+	*dalec.Spec
+	Target string
+}
+
+func (w *controlWrapper) Architecture() string {
+	if w.NoArch {
+		return "all"
+	}
+	return "linux-any"
+}
+
+func injectConstraints(in []string, deps map[string]dalec.PackageConstraints) {
```

Comment:
Can you change this function to return a new slice with the constraints attached to each package instead of mutating the passed in one? That reads more clearly to me. If so, it could be renamed to something like `appendConstraints`?

Reply:
Classic case of "wtf was I thinking when I wrote this" moment.
All calls to `injectConstraints` make the same `injectConstraints(slices.Sort(maps.Keys(deps)), deps))`

Reply:
Done.